chrysom/listenerClient_test.go

```diff
@@ -101,56 +209,60 @@ func newStartStopClient(includeListener bool) (*ListenerClient, func(), error) {
 		rw.Write(getItemsValidPayload())
 	}))
 
-	var listener Listener
+	var listener ListenerInterface
 	if includeListener {
 		listener = mockListener
 	}
-	client, err := NewListenerClient(listener,
-		func(context.Context) *zap.Logger { return zap.NewNop() },
-		func(context.Context, *zap.Logger) context.Context { return context.Background() },
-		time.Millisecond*200, pollsTotalCounter, &BasicClient{client: http.DefaultClient})
+	anclaClient, err := NewBasicClient(ClientOptions{
+		StoreBaseURL("https://example.com"),
+		Bucket("bucket-name"),
+		GetClientLogger(func(context.Context) *zap.Logger { return zap.NewNop() }),
+		HTTPClient(http.DefaultClient),
+	})
+	if err != nil {
+		return nil, func() {}, err
+	}
+
+	listenerClient, err := NewListenerClient(pollsTotalCounter,
+		ListenerOptions{
+			PullInterval(time.Millisecond * 200),
+			reader(anclaClient),
+			Listener(listener),
+			GetListenerLogger(func(context.Context) *zap.Logger { return zap.NewNop() }),
```

Comment:
These are good examples of where passing a `nil` func into these and having the option just return no-op functions like this are valuable for everyone.  Both line 231 and 232.